command_runner/runner.go

```diff
@@ -29,14 +30,22 @@ type Runner interface {
 	RunSubCmdByName(cmdName string, subCommand string, c *cli.Context) (err error)
 }
 
+type Factory interface {
```

Comment:
Hm, that suggests we maybe need a refactor to bring commonly required code into a lower level that will break the circular dependency. I think it would be really nice to avoid the duplication (triplication!) of interfaces.